backend_sndio/backend_sndio.c

```diff
@@ -0,0 +1,220 @@
+/*-
+ * Copyright (c) 2021 Tim Creech <tcreech@tcreech.com>. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#include <err.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sys/queue.h>
+#include <sys/soundcard.h>
+#include <sndio.h>
+
+#include "../virtual_int.h"
+#include "../virtual_backend.h"
+
+struct sio_ctx {
+	struct sio_hdl* hdl;
+	long long realpos;
```

Comment:
These fields are no longer used, so you can remove them?

Reply:
I can remove the fields, this struct, and the whole "onmove" callback arrangement. I only have these in here in hopes that I'll eventually find a way to make the computed delay available to OSS clients. Video players like "mpv" seem to use SNDCTL_DSP_GETODELAY to achieve audio/video sync. I think cubeb in firefox might do the same for OSS.

Would you prefer I rip all this stuff out, or leave it as a starting point for follow-up work along those lines?

Reply:
Yes. Just rip it out and I'll take the rest.

Reply:
OK, no problem. Will do.